File: .NET-app/MorkosiaPrepaLeague/Models/Core/ConcreteActivity.cs
namespace MorkosiaPrepaLeague.Models.Core
{
    public class ConcreteActivity : CosmosDocumentBase
    {
        public string DocumentType { get; } = "ConcreteActivity";

        public string TransferWindowId { get; set; } = string.Empty;

        public string ActivityTypeId { get; set; } = string.Empty;

        public DateOnly Date { get; set; }

        public int? OverridePoints { get; set; }

        public List<string> ParticipantIds { get; set; } = new();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: src/app/models/concrete_activity.py
from datetime import date, datetime, timezone
from typing import List, Optional

from beanie import Indexed
from pydantic import Field

from app.models.base import DocumentBase


class ConcreteActivity(DocumentBase):
    document_type: str = "ConcreteActivity"
    transfer_window_id: Indexed(str) = ""  # type: ignore[valid-type]
    activity_type_id: str = ""
    date: date
    override_points: Optional[int] = None
    participant_ids: List[str] = []
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "concrete_activities"
